channel-api/src/main/java/com/synopsys/integration/alert/channel/api/issue/convert/IssueVulnerabilityDetailsConverter.java

```diff
@@ -0,0 +1,100 @@
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.api.issue.convert;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+import com.synopsys.integration.alert.channel.api.convert.ChannelMessageFormatter;
+import com.synopsys.integration.alert.channel.api.issue.model.IssueVulnerabilityDetails;
+import com.synopsys.integration.alert.channel.api.issue.model.IssueVulnerabilityModel;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+import com.synopsys.integration.alert.processor.api.extract.model.project.ComponentConcernSeverity;
+
+public class IssueVulnerabilityDetailsConverter {
+    private static final String SECTION_LABEL_VULNERABILITIES = "Vulnerabilities: ";
+    private static final String OP_PARTICIPLE_ADDED = "Added: ";
+    private static final String OP_PARTICIPLE_UPDATED = "Updated: ";
+    private static final String OP_PARTICIPLE_DELETED = "Deleted: ";
+    private static final String LABEL_SEVERITY = "Severity: ";
+
+    private final ChannelMessageFormatter formatter;
+
+    public IssueVulnerabilityDetailsConverter(ChannelMessageFormatter formatter) {
+        this.formatter = formatter;
+    }
+
+    public List<String> createVulnerabilityDetailsSectionPieces(IssueVulnerabilityDetails vulnerabilityDetails) {
+        List<String> vulnDetailsSectionPieces = new LinkedList<>();
+
+        vulnDetailsSectionPieces.add(formatter.encode(SECTION_LABEL_VULNERABILITIES));
+
+        List<String> addedSectionPieces = createIssueVulnerabilityCollectionSectionPieces(OP_PARTICIPLE_ADDED, vulnerabilityDetails.getVulnerabilitiesAdded());
+        vulnDetailsSectionPieces.addAll(addedSectionPieces);
+
+        List<String> updatedSectionPieces = createIssueVulnerabilityCollectionSectionPieces(OP_PARTICIPLE_UPDATED, vulnerabilityDetails.getVulnerabilitiesUpdated());
+        vulnDetailsSectionPieces.addAll(updatedSectionPieces);
+
+        List<String> deletedSectionPieces = createIssueVulnerabilityCollectionSectionPieces(OP_PARTICIPLE_DELETED, vulnerabilityDetails.getVulnerabilitiesDeleted());
+        vulnDetailsSectionPieces.addAll(deletedSectionPieces);
```

Comment:
If you pass vulnDetailsSectionPieces into  createIssueVulnerabilityCollectionSectionPieces you could change all of this into 3 lines of code and perform the addAll in the private method

Reply:
I originally did this, but I don't feel that it is as expressive.